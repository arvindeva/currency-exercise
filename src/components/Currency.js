import React from 'react';

const Currency = props => {
  return (
    <div className="ui segment">
      <h1>{props.currency} - {props.usdInput * props.exchangeRate}</h1>
      <p>1 USD = {props.exchangeRate} {props.currency}</p>
    </div>
  );
};

export default Currency;
