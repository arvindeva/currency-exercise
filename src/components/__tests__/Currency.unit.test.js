import React from 'react';
import { shallow, mount } from 'enzyme';

import Currency from '../Currency';

describe('Currency component tests', () => {
  test('renders Currency', () => {
    const wrapper = shallow(<Currency />);
    expect(wrapper).toMatchSnapshot();
  });

  test('calls onClick event on click of the minus sign', () => {
    const onRemoveClick = jest.fn();
    let wrapper = mount(<Currency onRemoveClick={onRemoveClick} />);
    wrapper
      .find('.minus')
      .first()
      .simulate('click');
    expect(onRemoveClick).toHaveBeenCalled();
  });
});
