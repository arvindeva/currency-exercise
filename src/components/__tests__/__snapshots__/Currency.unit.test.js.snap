// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currenct component tests renders Currency 1`] = `
<styled.div>
  <div
    className="left"
  >
    <div
      className="top"
    >
      <div>
        <h1 />
      </div>
      <div>
        <h1>
           
          NaN
        </h1>
      </div>
    </div>
    <div
      className="middle"
    >
      <p>
         - 
      </p>
    </div>
    <div
      className="bottom"
    >
      <p>
        1 USD =
         
         
      </p>
    </div>
  </div>
  <div
    className="right"
  >
    <Icon
      as="i"
      className="minus"
      name="minus square outline"
      onClick={[Function]}
      size="big"
    />
  </div>
</styled.div>
`;
